src/libraries/System.Private.Xml/src/System/Xml/Serialization/TypeExtensions.cs

```diff
@@ -4,13 +4,16 @@
 namespace System.Xml.Serialization
 {
     using System;
+    using System.Diagnostics.CodeAnalysis;
     using System.Reflection;
 
     internal static class TypeExtensions
     {
         private const string ImplicitCastOperatorName = "op_Implicit";
 
-        public static bool TryConvertTo(this Type targetType, object? data, out object? returnValue)
+        public static bool TryConvertTo(
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods
+                | DynamicallyAccessedMemberTypes.NonPublicMethods)] this Type targetType, object? data, out object? returnValue)
```

Comment:
I just see `Public` used below. Do we really need `NonPublicMethods` on here?

Reply:
Hmm, the linker seems to think so, but I concur, this looks like a bug.

Reply:
I removed the nonPublicMethods and the linker was no longer warning so if it was a bug it looks like it is fixed now.